File: dwcj-engine/src/main/java/org/dwcj/component/html/elements/H2.java
package org.dwcj.component.html.elements;

import org.dwcj.component.Component;
import org.dwcj.component.element.annotation.NodeName;
import org.dwcj.component.html.HtmlComponentContainer;

/**
 * Component representing a {@code h2} element.
 *
 * @see <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements">HTML h2
 *      Tag</a>
 *
 * @author Hyyan Abo Fakher
 * @since 23.06
 */
// We're purposefully ignoring the deep inheritance warning here because we've designed our class
// hierarchy to meet the unique requirements of our UI framework. This design closely aligns with
// our framework's specific goals and emphasizes the need for caution when considering any changes.
//
// Any changes to the inheritance structure should be thoughtfully evaluated in the context of our
// framework's needs. The current structure is essential for meeting those needs.
@SuppressWarnings("squid:S110")
@NodeName("h2")
public class H2 extends HtmlComponentContainer<H2> {

  /**
   * Creates a new empty heading.
   */
  public H2() {
    super();
  }

  /**
   * Creates a new heading with the given text.
   *
   * @param text the text
   */
  public H2(String text) {
    super();
    setText(text);
  }

  /**
   * Creates a new heading with the given child components.
   *
   * @param components the child components
   */
  public H2(Component... components) {
    super(components);
  }
}

File: webforj-kotlin/src/main/kotlin/com/webforj/kotlin/dsl/component/html/elements/H2.kt
package com.webforj.kotlin.dsl.component.html.elements

import com.webforj.component.html.elements.H2
import com.webforj.concern.HasComponents
import com.webforj.kotlin.dsl.WebforjDsl
import com.webforj.kotlin.dsl.init

/**
 * Creates an `H2` with an optional [text].
 * ```
 * ... {
 *   H2() // Empty h2 element
 *   H2("text") // h2 element with text
 * }
 */
fun @WebforjDsl HasComponents.h2(text: String? = null, block: @WebforjDsl H2.() -> Unit = {}): H2 {
    val h2 = text?.let { H2(text) } ?: H2()
    return init(h2, block)
}
